components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/JITProvisioningPostAuthenticationHandler.java

```diff
@@ -210,7 +209,6 @@ private PostAuthnHandlerFlowStatus handleResponseFlow(HttpServletRequest request
                     if (context.getProperty(FrameworkConstants.CHANGING_USERNAME_ALLOWED) != null) {
                         username = request.getParameter(FrameworkConstants.USERNAME);
                     }
-                    isUsernameExists(context, username);
```

Comment:
Why are we removing these validations? Seems like this will make behavioral change in this flow.  Have we capture this validation in a previous step?

Reply:
These validations being present here causes the federated authentication flow to break upon validation failure. This validation is now done through the DefaultProvisioningHandler with the assistance of the introduced configuration to allow the federated authentication flow to continue even if the JIT provisioning fails due to username already being used. 
Updated the PR comment with this information.